layers/core_checks/cc_copy_blit_resolve.cpp

```diff
@@ -2452,16 +2454,19 @@ bool CoreChecks::ValidateImageBufferCopyMemoryOverlap(const CMD_BUFFER_STATE &cb
             DispatchGetImageSubresourceLayout(device, image_state.image(), &isr, &srl);
             if (image_state.createInfo.arrayLayers == 1) srl.arrayPitch = 0;
             if (image_state.createInfo.imageType != VK_IMAGE_TYPE_3D) srl.depthPitch = 0;
-            image_offset = (region.imageSubresource.baseArrayLayer * srl.arrayPitch) + (region.imageOffset.x * element_size) +
-                           (region.imageOffset.y * srl.rowPitch) + (region.imageOffset.z * srl.depthPitch) + srl.offset;
+            image_offset = (region.imageSubresource.baseArrayLayer * srl.arrayPitch) +
+                           static_cast<VkDeviceSize>((region.imageOffset.x * texel_size)) + (region.imageOffset.y * srl.rowPitch) +
+                           (region.imageOffset.z * srl.depthPitch) + srl.offset;
         } else {
-            image_offset = region.imageOffset.x * region.imageOffset.y * region.imageOffset.z * element_size;
+            image_offset =
+                static_cast<VkDeviceSize>(region.imageOffset.x * region.imageOffset.y * region.imageOffset.z * texel_size);
```

Comment:
```suggestion
                static_cast<VkDeviceSize>(region.imageOffset.x) * region.imageOffset.y * region.imageOffset.z * texel_size;
```
(Same assumption here)

Reply:
It should be floating point operations cast to uint64_t